Adds interactive mode as default. Closes #5181

Adam-it opened the pull request:
## 🔗 Linked Issue

Closes #5181

## 📷 Result

After this change, every command run will automatically ask (prompt) for 
- not specified required options,
- not specified one option among a set of required (options set)
- when the command output will give more than one result, and only single value was expected, it will ask to pick the correct result

![image](https://github.com/pnp/cli-microsoft365/assets/58668583/5ef95cd0-ac35-4744-a750-534f16d82301)

waldekmastykarz commented on src/m365/aad/commands/app/app-get.spec.ts:
```diff
@@ -145,21 +145,53 @@ describe(commands.APP_GET, () => {
   });
 
   it('fails validation if appId and objectId specified', async () => {
+    sinon.stub(cli, 'getSettingWithDefaultValue').callsFake((settingName, defaultValue) => {
```
If we provide all required options in the `command.validate()` call below, then this shouldn't be necessary, right? And if the logic changes, and we add a required option that's missing, then the test will fail with a timeout. I wonder if that wouldn't be sufficient and let us avoid adding all this code to all validation tests.

If we don't want timeouts, then we should consider adding this stub in `before` and then cleaning it up in `after` to avoid repeating it. Since the logic to test required options is centralized and not a part of command, we won't need to unstub `cli.getSettingWithDefaultValue()` for specific tests, so it would save us some work to stub it once

waldekmastykarz replied:
@Adam-it clarified offline that the current approach is necessary to support other test cases like disambiguation prompting. We'll this as-is

Adam-it commented:
rebased ✅

waldekmastykarz commented:
Could you please re-open the PR against main? Because I removed the v7 branch, the PR got closed. Sorry for that, it wasn't my intention

Adam-it commented:
> Could you please re-open the PR against main? Because I removed the v7 branch, the PR got closed. Sorry for that, it wasn't my intention

sure. Moved to https://github.com/pnp/cli-microsoft365/pull/5442
